api/src/main/java/org/openmrs/ConceptReferenceRange.java

```diff
@@ -0,0 +1,423 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.codehaus.jackson.annotate.JsonIgnore;
+import org.hibernate.envers.Audited;
+import org.hibernate.search.annotations.DocumentId;
+import org.hibernate.search.annotations.Field;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.util.Date;
+
+/**
+ * A concept reference range is typically a range of a concept with a certain criteria. The criteria is used to evaluate
+ * if certain attributes of a patient meet the specified criteria before applying the range. 
+ * For example, if criteria is of factor age(say age between 1-5), then the ranges only apply for this age group.
+ *
+ * @since 2.7.0
+ */
+@Entity
+@Table(name = "concept_reference_range")
+@Audited
+public class ConceptReferenceRange extends BaseOpenmrsObject implements Auditable, Retireable {
```

Comment:
Is there a reason why you did not extend Concept or ConceptNumeric?

Reply:
The main reason is that we don't want inter-dependency. Yes we want backward compatibility but not convolution of objects such that later when we decide to refactor or do away with ConceptNumeric, it will be difficult to do so.

Reply:
It does not look nice to duplicate the many fields which are already in ConceptNumeric. If we were to later on get rid of it, which i do not see happening in the nearby future, we would just copy them into this class.

Reply:
Okay, I will extend ConceptNumeric

Reply:
@dkayiwa I considered extending ConceptNumeric/Concept, but it’s not a good practice since we would inherit unnecessary attributes like conceptSets, descriptions, etc. To achieve single responsibility, could we separate these entities? We could create a more focused abstract class for inheritance instead.

Reply:
Sounds fair enough. But i would not touch those existing classes, at least for now. 😊

Reply:
This is well noted.